clients/lsp-javascript.el

```diff
@@ -727,6 +819,63 @@ Code's JavaScript and TypeScript support."
       (lsp)
       (lsp--info "Renamed '%s' to '%s'." name (file-name-nondirectory new)))))
 
+(defun lsp-javascript-update-inlay-hints (buffer)
+  (if (and (lsp-javascript-initialized?)
+           (eq buffer (current-buffer)))
+      (lsp-request-async
+       "typescript/inlayHints"
+       (lsp-make-javascript-inlay-hints-params
+        :text-document (lsp--text-document-identifier))
+       (lambda (res)
+         (remove-overlays (point-min) (point-max) 'lsp-javascript-inlay-hint t)
```

Comment:
I guess you have copied this from the rust implementation but here you should use `lsp--remove-overlays` to ensure it will work in narrowed buffers.

Reply:
Done, thanks.